### PR TITLE
Added test framework files.

### PR DESCRIPTION
**Pull request by kkleimenova**
@SergiiNozhka Please review added data.

### CODE REVIEW

File: src/test/java/page/structure/Page.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        do {
+            js = (JavascriptExecutor) driver;
+            pageLoadStatus = (String)js.executeScript("return document.readyState");
+        } while ( !pageLoadStatus.equals("complete") );
+        return page;
+    }
+
+    public Page(){
+    	this.driver = getDriver();
```
Comment (SergiiNozhka) [inline src/test/java/page/structure/Page.java]: Please fix indentations.

File: src/test/java/page/structure/ResultsPage.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.google.common.base.Verify;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.FindBy;
+import org.testng.Assert;
+
+public class ResultsPage extends Page{
+
+    public WebDriver driver;
+
```
Comment (SergiiNozhka) [inline src/test/java/page/structure/ResultsPage.java]: Remove blank lines.

File: src/test/java/tests/BaseClass.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    public static WebDriver getDriver() {
+        return driver;
+    }
+
+    @DataProvider(name = "Authentication")
+
+    public static Object[][] credentials() {
+
+        return new Object[][] { { "testuser_1", "Test@123" }, { "testuser_1", "Test@123" }};
```
Comment (SergiiNozhka) [inline src/test/java/tests/BaseClass.java]: Suggest move to config file and here implement reading from introduced config.

File: src/test/java/tools/CustomITestListener.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        Reporter.log("<img src=\"file:///" + destFile + "\" alt=\"\"/><br />");
+
+    }
+
+    private void deleteFilesFromDir(String path) {
+        File file = new File(path);
+        File[] files = file.listFiles();
+        for (File f : files) {
+            if (f.isFile() && f.exists()) ;
+            {
```
Comment (SergiiNozhka) [inline src/test/java/tools/CustomITestListener.java]: Please follow the same code-style across all files.

File: src/test/java/tools/CustomITestListener.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                f.delete();
+            }
+            {
+                System.out.println("ER: Can't delete files from path:" + path + " due to open or error.");
```
Comment (SergiiNozhka) [inline src/test/java/tools/CustomITestListener.java]: Replace sout with log.

File: src/test/java/tools/CustomITestListener.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    public void onTestSuccess(ITestResult result) {
+
+    }
+
+    public void onTestFailure(ITestResult result) {
+        WebDriver driver = ((BaseClass) result.getInstance()).getDriver();
+
+        File imageFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
+        String failureImageFileName = result.getMethod().getMethodName() + new SimpleDateFormat("__MM-dd-yyyy__HH-ss").format(new GregorianCalendar().getTime()) + ".png";
```
Comment (SergiiNozhka) [inline src/test/java/tools/CustomITestListener.java]: Suggest move format to constant.

File: src/test/java/page/structure/Page.java (hunk 1 above)
Comment (kkleimenova) [reply src/test/java/page/structure/Page.java]: Done.

File: src/test/java/page/structure/ResultsPage.java (hunk 2 above)
Comment (kkleimenova) [reply src/test/java/page/structure/ResultsPage.java]: Why do I need to remove this blank lines?

File: src/test/java/tools/CustomITestListener.java (hunk 4 above)
Comment (kkleimenova) [reply src/test/java/tools/CustomITestListener.java]: I don't understand this.

File: src/test/java/page/structure/ResultsPage.java (hunk 2 above)
Comment (SergiiNozhka) [reply src/test/java/page/structure/ResultsPage.java]: Please read code-style doc

File: src/test/java/page/structure/MainPage.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        this.driver = driver;
+    }
+
+    public MainPage openMainPage() throws IOException {
+        System.out.print(Constants.URL);
+        driver.get(Constants.URL);
+        return this;
+    }
+
+//    public MainPage enterUsername() {
```
Comment (SergiiNozhka) [inline src/test/java/page/structure/MainPage.java]: Remove commented or uncomment.

File: src/test/java/tools/CustomITestListener.java (hunk 4 above)
Comment (SergiiNozhka) [reply src/test/java/tools/CustomITestListener.java]: Please read code-style doc

File: src/test/java/page/structure/ResultsPage.java (hunk 2 above)
Comment (kkleimenova) [reply src/test/java/page/structure/ResultsPage.java]: Done.

File: src/test/java/tools/CustomITestListener.java (hunk 4 above)
Comment (kkleimenova) [reply src/test/java/tools/CustomITestListener.java]: Done.

File: src/test/java/tools/CustomITestListener.java (hunk 5 above)
Comment (kkleimenova) [reply src/test/java/tools/CustomITestListener.java]: Done.

File: src/test/java/tools/CustomITestListener.java (hunk 6 above)
Comment (kkleimenova) [reply src/test/java/tools/CustomITestListener.java]: No need, course it is used only in one place.

File: src/test/java/page/structure/MainPage.java (hunk 7 above)
Comment (kkleimenova) [reply src/test/java/page/structure/MainPage.java]: Done.

File: src/test/java/tests/BaseClass.java (hunk 3 above)
Comment (kkleimenova) [reply src/test/java/tests/BaseClass.java]: Done.

**Pull request merged**